test/fixtures/api_actions.yml

```diff
@@ -124,3 +124,75 @@ create_api_action_plugin_bishop_monitoring_email:
   include_api_resource_data: true
   email: test@restarone.com
   api_namespace: monitoring_target_incident
+
+custom_api_action_on_timesheet_request:
+  type: CreateApiAction
+  action_type: custom_action
+  include_api_resource_data: true
+  payload_mapping:
+  api_namespace: timesheet_request
+  method_definition: |
+    # get time tracker namespace
+    time_tracker = ApiNamespace.find_by(name: 'time_tracker')
+    
+    format = "%Y-%m-%d"
+    if api_resource.properties['current_month'] == "true"
+      api_resource.update(properties: api_resource.properties.merge({ start_date: Date.today.beginning_of_month.strftime(format), end_date: Date.today.end_of_month.strftime(format) }))
+    end
+    
+    # get time tracker entries for the current user within the specified time
+    start_date = Date.parse(api_resource.properties['start_date'])
+    end_date = Date.parse(api_resource.properties['end_date'])
+    # email = current_user.email
+    email = "test@restarone.solutions"
+    current_user_time_tracker_entries = time_tracker.api_resources.filter { |api_resource| api_resource.properties['email_address'] == email && api_resource.created_at.to_date >= start_date && api_resource.created_at.to_date <= end_date }
```

Comment:
please use a `.where` query here and avoid loading all resources

Reply:
yup Don, fixed the method